addons/website_sale/static/src/js/website_sale.editor.js

```diff
@@ -186,14 +191,129 @@ options.registry.WebsiteSaleProductsItem = options.Class.extend({
     /**
      * @see this.selectClass for params
      */
-    style: function (previewMode, widgetValue, params) {
-        this._rpc({
-            route: '/shop/change_styles',
-            params: {
-                'id': this.productTemplateID,
-                'style_id': params.possibleValues[params.possibleValues.length - 1],
-            },
-        });
+    async setRibbon(previewMode, widgetValue, params) {
+        this.$target[0].dataset.ribbonId = widgetValue;
+        if (this.ribbons[widgetValue]) {
+            await this._rpc({
+                route: '/shop/set_ribbon',
+                params: {
+                    'product_tmpl_id': this.productTemplateID,
+                    'ribbon_id': widgetValue || false,
+                },
+            });
+        }
+        const ribbon = this.ribbons[widgetValue] || {name: '', color: 'rgba(0,0,0,0)', text_color: ''};
+        const $ribbons = $(`[data-ribbon-id="${widgetValue}"] .o_wribbon`);
+        $ribbons.text(ribbon.name);
+        $ribbons.attr('class', 'o_wribbon');
+
+        if (ribbon.color.startsWith('bg-')) {
+            $ribbons.addClass(ribbon.color);
+            $ribbons.css('background-color', '');
+        } else {
+            $ribbons.css('background-color', ribbon.color);
+        }
+        $ribbons.css('color', ribbon.text_color);
+
+        if (!this.ribbons[widgetValue]) {
+            $(`[data-ribbon-id="${widgetValue}"]`).each((index, product) => delete product.dataset.ribbonId);
+        }
+    },
+    /**
+     * @see this.selectClass for params
+     */
+    editRibbon(previewMode, widgetValue, params) {
+        this.saveMethod = 'write';
+        this.ribbonColor = this.$('.o_wribbon').css('background-color');
+        this.ribbonTextColor = this.$('.o_wribbon').css('color');
+        this.ribbonName = this.$('.o_wribbon').text();
+        this._toggleEditingUI(true);
+    },
+    /**
+     * @see this.selectClass for params
+     */
+    createRibbon(previewMode, widgetValue, params) {
+        this.saveMethod = 'create';
+        this.ribbonName = 'Ribbon text';
+        this.ribbonColor = '#ff9c00';
+        this._toggleEditingUI(true);
+    },
```

Comment:
Not sure I am a fan of using the option system for stuff like that (even `saveRibbon` and `deleteRibbon` are questionable but kinda necessary for the `async` parts)... but I guess it is ok. No real argument to avoid it... In this case, I would have rendered the 2 buttons in JS and attached event handlers through the event properties. Like a normal widget.

Reply:
Not sure what you mean here with "stuff like that"

Reply:
Toggling another part of the UI instead of editing the DOM like options are meant to, but I guess it is not really different from "opening a modal to edit something".